app/assets/javascripts/utils/course_date_utils.js

```diff
@@ -93,49 +86,42 @@ const CourseDateUtils = {
   MAX_MONTHS: 13,
 
   courseTooLong(course) {
-    return moment(course.end).diff(moment(course.start), 'months') > this.MAX_MONTHS;
+    return differenceInMonths(toDate(course.end), toDate(course.start)) > this.MAX_MONTHS;
   },
 
   moreWeeksThanAvailable(course, weeks, exceptions) {
     if (!weeks || !weeks.length) { return false; }
-    const nonBlackoutWeeks = filter(this.weekMeetings(this.meetings(course), course, exceptions), mtg => mtg.length > 0);
+    const nonBlackoutWeeks = filter(this.weekMeetings(course, exceptions), mtg => mtg.length > 0);
     return weeks.length > nonBlackoutWeeks.length;
   },
 
   wouldCreateBlackoutWeek(course, day, exceptions) {
-    const selectedDay = moment(day);
+    const selectedDay = toDate(day);
     let noMeetingsThisWeek = true;
     [0, 1, 2, 3, 4, 5, 6].forEach((i) => {
-      const wkDay = selectedDay.day(0).add(i, 'days').format('YYYYMMDD');
+      const wkDay = format(addDays(startOfWeek(selectedDay), i), 'yyyyMMdd');
       if (this.courseMeets(course.weekdays, i, wkDay, exceptions.join(','))) { return noMeetingsThisWeek = false; }
     });
     return noMeetingsThisWeek;
   },
 
   weeksBeforeTimeline(course) {
-    const courseStart = moment(course.start).startOf('week');
-    const timelineStart = moment(course.timeline_start).startOf('week');
-    return timelineStart.diff(courseStart, 'week');
+    const courseStart = startOfWeek(toDate(course.start));
+    const timelineStart = startOfWeek(toDate(course.timeline_start));
+    return differenceInWeeks(timelineStart, courseStart);
   },
 
   // Returns array describing weekday meetings for each week
   // Ex: [["Sunday (01/09)"], ["Sunday (01/16)", "Wednesday (01/19)", "Thursday (01/20)"], []]
-  weekMeetings(recurrence, course, exceptions) {
-    if (!recurrence) { return []; }
-    const weekEnd = recurrence.endDate();
-    weekEnd.day(6);
-    let weekStart = recurrence.startDate();
-    const firstWeekStart = recurrence.startDate().day();
-    weekStart.day(0);
-    const courseWeeks = Math.ceil(weekEnd.diff(weekStart, 'weeks', true));
-    if (!recurrence.rules || recurrence.rules[0].measure !== 'daysOfWeek') {
-      return [];
-    }
-
+  weekMeetings(course, exceptions) {
```

Comment:
Can you explain more about the change to this function and why the first argument (and the `meetings` function) are not needed?